sqlmesh/core/snapshot/definition.py

```diff
@@ -110,11 +110,17 @@ def snapshot_id(self) -> SnapshotId:
         """Helper method to return self."""
         return self
 
+    def __lt__(self, other: SnapshotId) -> bool:
+        return self.name < other.name
+
 
 class SnapshotNameVersion(PydanticModel, frozen=True):
     name: str
     version: str
 
+    def __str__(self) -> str:
+        return "__".join([self.name, self.version])
```

Comment:
Why is this needed?

Reply:
Example of usage: https://github.com/TobikoData/sqlmesh/blob/7c689c680cf29541a1a546414a52a78db8de850c/sqlmesh/core/state_sync/engine_adapter.py#L505

Reply:
isn't it already hashable?

Reply:
Yeah, I'd expect `frozen=True` to ensure that

Reply:
Even then, pydantic provides a default `__str__` implementation already:
```
In [3]: str(SnapshotNameVersion(name="test_name", version="test_version"))
Out[3]: "name='test_name' version='test_version'"
```

Reply:
Ah great points. Will fix.